### ISSUE TITLE
JSON to float conversion issue

### ISSUE THREAD

**Issue by riyafa** [ballerina-platform/ballerina-lang#12363]
**Description:**
The following code prints false:
```
import ballerina/io;
public function main() {
    json operationReq = {"firstNumber": 10, "secondNumber": 200.1, "operation": "add"};
    var input = float.convert(operationReq.firstNumber);
    io:println(input is float);
}
```

Whereas this conversion should return true on successful conversion.

**Comment by riyafa** [comment]
Now the above code does not compile and the following compilation error occurs:
`error: .::create.bal:4:17: function invocation on type 'typedesc' is not supported`
If this is the expected behavior this issue can be closed

**Comment by riyafa** [comment]
Now even this does not compile:
```
import ballerina/io;
public function main() {
    json operationReq = {"firstNumber": 10, "secondNumber": 200.1, "operation": "add"};
    int val = operationReq.firstNumber;
    var input = float.create(val);
    io:println(input is float);
}
```
How am I going to get a float out of this json?

**Comment by SupunS** [comment]
Fixed with #12515
